### ISSUE TITLE
Diagram to show input/output

### ISSUE THREAD

**Issue by sonukapoor** [rangle/angular-training-slides#492]
How do you feel about a diagram similar to the one below to show how the input / output workflow?

![image](https://cloud.githubusercontent.com/assets/59734/25622137/7cded022-2f21-11e7-80ef-973f8e58003f.png)

@gvwilson @rpdasilva @jamesbs @mdegani

**Comment by gvwilson** [comment]
I like it!

**Comment by rpdasilva** [comment]
Works for me - Good way to show the relationship between template syntax and component classes

**Comment by sonukapoor** [comment]
Let me know if you want any changes to this. I have updated the image.

**Comment by jamesbs** [comment]
It would be nice if there were some way to draw the arrows in live. There's a bit of information overload with arrows going in different directions

**Comment by sonukapoor** [comment]
I know what you mean. Maybe we can display multiple images going from one 1 arrow into 4, while explaining what each of them do. What do you think about that?

**Comment by mdegani** [comment]
I like the diagram. It's not something you can just glance at though, takes a while to follow all of the arrows. Maybe the red is a bit stark and makes it look busier than it actually is.

What if the arrows pointed _from_ the thingsToDo assignment `thingsToDo = [...` to the template to illustrate that data is flowing from there to the input in the ToDoListComponent.  Also, maybe the `@Output...` could point to the template? There are 5 arrows, but there are only two things happening: data flowing down and data being emitted upwards.

**Comment by mdegani** [comment]
Actually, you could even number each arrow starting with the output and show each step that is triggered by the `newItem` event and leading to the addition of the item to the list.

**Comment by sonukapoor** [comment]
Yes. I was thinking about numbering them as well. I am going to try to create 4 different image, starting with 1 arrow and adding another one with each image. The end image should look like the original one. Maybe that will be easier to explain and could appear less complicated.
